autoload/ale/hover.vim

```diff
@@ -160,10 +160,11 @@ function! ale#hover#ParseLSPResult(contents) abort
                 let l:language = s:ConvertLanguageName(l:language)
 
                 if !empty(l:language)
-                    let l:includes[l:language] = printf(
-                    \   'syntax/%s.vim',
-                    \   l:language,
-                    \)
+                    let l:syntax_file = printf('syntax/%s.vim', l:language)
+
+                    if !empty(globpath(&runtimepath, l:syntax_file))
```

Comment:
Let's write a script function `s:SyntaxFileExists` and store the result in script variable so we only run `globpath`  the first time we encounter a particular language. That will stop us scanning the filesystem each time we want to show a message. We wouldn't show highlights for new syntax files until Vim is restarted, but I think that's a low cost to pay.

Reply:
Done: https://github.com/dense-analysis/ale/compare/36ace32666f2385a580c3b04193a6dc60e183cb5..04c123524426955bc8992aaf9db805f46746cbe5